### PR TITLE
[daikin] Support BRP069A81 adapter

### LINKED ISSUE #15375: [daikin] Bug / Missing value Fanrate f_rate and configurable value for AUTO mode
Hi,

I am currently setting up two Daikin AirCons. When testing different setup and configs, I found two cases that cause problems. One seems to be rather a bug, the other one is a feature request.

Setup
I have two things configured:

```
/* stylish wall mount device*/
daikin:ac_unit:sleeping_room_ac "Klimaanlage Schlafzimmer" [
  host="192.x.y.9"
]

/* BRP069A81 */
daikin:ac_unit:living_room_ac "Klimaanlage Wohnzimmer" [
  host="192.y.x.8"
]
```
**BUG: DEHUMIDIFIER and Fan Rate**
When setting the mode to DEHUMIDIFIER (either directly via http-API or the binding), the daikin sets stemp=-- and f_rate=-. While stemp=-- does not cause any issues and is interpreted correctly, the fact that is set to f_rate=- causes a debug message:

```
2023-06-24 23:10:27.937 [DEBUG] [g.daikin.internal.api.Enums$FanSpeed] - Unexpected FanSpeed value of “-”
```
The API reply for http://192.x.y.8/aircon/get_control_info in DEHUMIDIFIER Mode is
`ret=OK,pow=0,mode=2,stemp=–,shum=–,dt1=25.0,dt2=–,dt3=25.0,dt4=25.0,alert=0,adv=,f_rate=-,dfr1=A,dfr2=-,dfr3=A,dfr4=A,dfr6=3`
It would be great if the “-” could be considered as valid value for f_rate.

**Feature: Correct Mode Value for**
As written above, I am using a BRP069A81 adapter to connect the AirCon. I know that this is formally not supported but it works almost well.
While everything works fine, there is a slight difference in how the mode value is interpreted. While my other Daikin uses the value 0 for AUTO mode (which works fine with the binding), the BRP069A81 uses the value 1 instead. This is a value the binding does not understand and as 0 is set instead of the required 1, the Aircon cannot be controlled when it is in AUTO mode.
Here, it would be great to have an optional configuration option in the daikin:ac_unit:* thing to use the value 1 instead of 0 for mode = AUTO.

Happy to provide more details & discuss.

Thanks, Lui

### PR DESCRIPTION
**Pull request by jimtng**
Daikin BRP069A81 adapter uses mode=1 for its AUTO mode. This PR makes the binding adaptively uses the correct mode=0 or mode=1 depending on what the adapter reports back.

Resolve #15375

### CODE REVIEW

File: bundles/org.openhab.binding.daikin/README.md [hunk 1]
```diff
@@ -1,12 +1,16 @@
 # Daikin Binding
 
 The Daikin binding allows you to control your Daikin air conditioning units with openHAB.
-In order to do so, your Daikin air conditioning unit must have a BRP072A42, BRP072C42 or BRP15B61 WiFi adapter installed.
+
+In order to do so, your Daikin air conditioning unit must have a supported WiFi adapter installed.
```
Comment (jlaur) [inline bundles/org.openhab.binding.daikin/README.md:5]:
```suggestion
In order to do so, your Daikin air conditioning unit must have a supported Wi-Fi adapter installed.
```

File: bundles/org.openhab.binding.daikin/README.md [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+This may work with the older KRP series of wired adapters, but has not been tested with them.
 
 ## Supported Things
 
-Daikin air conditioning units with a BRP069B41, BRP072A42, BRP072C42 or BRP15B61 installed.
-This may work with the older KRP series of wired adapters, but has not been tested with them.
+| Thing             | Daikin Wifi Adapter Model                  |
```
Comment (jlaur) [inline bundles/org.openhab.binding.daikin/README.md:10]:
```suggestion
| Thing             | Daikin Wi-Fi Adapter Model                 |
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Two very minor comments added.

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java [hunk 3]
```diff
@@ -202,7 +205,21 @@ protected void changeMode(String mode) throws DaikinCommunicationException {
         }
         ControlInfo info = webTargets.getControlInfo();
         info.mode = newMode;
-        webTargets.setControlInfo(info);
+        if (autoModeValue.isPresent()) {
+            info.autoModeValue = autoModeValue.get();
+        }
+        boolean accepted = webTargets.setControlInfo(info);
+
+        // If mode=0 is not accepted try AUTO1 (mode=1)
```
Comment (lsiepel) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java:213]: Just double check question: 
As the Mode enum also has:
```
 AUTO7(7), // Some adapters may return 7 as auto (heating) 
```
and it is normalised to `Mode.AUTO` just like `Mode.AUTO1` it looks like it is missing from this addition.

Comment (jimtng) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java:213]: Yes, this is deliberate. From what we can gather:

- The current library uses mode=0, and that's why it remains the current default for AUTO
- We "suspect" that all models accept mode=1 (not fully tested against all supported models). A [post was made on the forum](https://community.openhab.org/t/daikin-binding-test-of-auto-mode-value-we-need-your-help/148919) to get others to test this. In the mean time, mode=0 remains the default for AUTO.
- BRP069A81 (for which this PR is intended), _only_ accepts mode=1. It does not accept mode=0 or mode=7
- Models that accept mode=0 "seem to" also accept mode=1 and mode=7 (not fully tested against all supported models)
- We "believe" that mode=0 and mode=7 have different meanings, and mode=1 is actually the most "AUTO" amongst the three codes. 
- There are no models, to our knowledge, that only accept mode=7 but not mode=0 or mode=1, so trying mode=7 is currently not necessary nor requested.

So in short:
- Use mode=0 as it was before, but if it doesn't work:
- Try mode=1 because that's the only one recognised by BRP069A81 (this PR)
- mode=7 is not attempted because BRP069A81 doesn't recognise it.

See #15375

Comment (lsiepel) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java:213]: Thanks for the extensive clarification.

**Review by jlaur — APPROVED** [review]
LGTM
